### PR TITLE
[flashlight-cuda] Update port

### PR DESCRIPTION
**Pull request by jacobkahn**
Same as https://github.com/microsoft/vcpkg/pull/17643, but for `flashlight-cuda`.

- #### What does your PR fix?  

Updates the port to include the fix to https://github.com/flashlight/flashlight/issues/574.

Correctly moves CMake assets in `[installed]/share/flashlight-cuda` into `[installed]/share/flashlight`

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

No change to supported triplets

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes

### CODE REVIEW

File: ports/flashlight-cuda/usage [hunk 1]
```diff
@@ -0,0 +1,11 @@
+The package flashlight-cuda provides CMake targets:
```
Comment (ras0219-msft) [inline ports/flashlight-cuda/usage:1]: Do our built-in automatic heuristics not work here?

Also, this is missing a call to `vcpkg_cmake_config_fixup()`: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake-config/vcpkg_cmake_config_fixup.md

```cmake
vcpkg_cmake_config_fixup()
```

Comment (jacobkahn) [reply ports/flashlight-cuda/usage:1]: Good point. Let me give that a try and see if this can mean we can remove the usage.

Comment (jacobkahn) [reply ports/flashlight-cuda/usage]: @ras0219 fixed using `vcpkg_fixup_cmake_targets(...)`. Ready for review

**Review by JonLiu1993 — APPROVED** [review]

File: versions/f-/flashlight-cuda.json [hunk 2]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "bd616669b5eb7413794d727b9909309ae686cd04",
+      "git-tree": "9147d15ccbb61b2de168a8cc78527341edf1540d",
+      "version": "0.3",
+      "port-version": 1
+    },
+    {
+      "git-tree": "56fb4895272b9e0d75b474e4b6caf941733fc495",
```
Comment (dan-shaw) [inline versions/f-/flashlight-cuda.json:9]: The `git-tree` reference of the previous version should be the same.

**Review by PhoebeHui — APPROVED** [review]
Thanks for the PR!

**Pull request merged**